plugins/out_pgsql/pgsql.c

```diff
@@ -0,0 +1,270 @@
+    ctx = NULL;
+}
+
+static int cb_pgsql_init(struct flb_output_instance *ins,
+                          struct flb_config *config, void *data)
+{
+
+    struct flb_pgsql_config *ctx;
+    size_t str_len;
+    PGresult *res;
+    char *query = NULL;
+    char *temp = NULL;
+    const char *tmp = NULL;
+
+    /* set default network configuration */
+    flb_output_net_default(FLB_PGSQL_HOST, FLB_PGSQL_PORT, ins);
+
+    ctx = flb_calloc(1, sizeof(struct flb_pgsql_config));
+    if (!ctx) {
+        flb_errno();
+        return -1;
+    }
+
+    ctx->db_hostname = flb_strdup(ins->host.name);
+    snprintf(ctx->db_port, sizeof(ctx->db_port), "%d", ins->host.port);
+
+
+    ctx->db_name = flb_output_get_property("database", ins);
+    if(!ctx->db_name) {
+        ctx->db_name = FLB_PGSQL_DBNAME;
+    }
+
+    tmp = flb_output_get_property("table", ins);
+    if(tmp) {
+        ctx->db_table = flb_sds_create(tmp);
+    } else {
+        ctx->db_table = flb_sds_create(FLB_PGSQL_TABLE);
+    }
+
+    if(!ctx->db_table) {
+        pgsql_conf_destroy(ctx);
+        return -1;
+    }
+
+    ctx->db_user = flb_output_get_property("user", ins);
+    if(!ctx->db_user) {
+        flb_warn("[out_pgsql] You didn't supply a valid user to connect, your current unix user will be used");
+    }
+
+    ctx->db_passwd = flb_output_get_property("password", ins);
+    if(!ctx->db_passwd) {
+        flb_warn("[out_pgsql] You didn't supply a password, you should use a password to authenticate against PostgreSQL");
+    }
+
+    tmp = flb_output_get_property("timestamp_key", ins);
+    if(tmp) {
+        ctx->timestamp_key = flb_sds_create(tmp);
+    } else {
+        ctx->timestamp_key = flb_sds_create(FLB_PGSQL_TIMESTAMP_KEY);
+    }
+
+    if(!ctx->timestamp_key) {
+        pgsql_conf_destroy(ctx);
+    }
+
+    flb_info("[out_pgsql] host=%s port=%s dbname=%s ...",
+             ctx->db_hostname, ctx->db_port, ctx->db_name);
+
+    ctx->conn = PQsetdbLogin(ctx->db_hostname,
+                             ctx->db_port,
+                             NULL, NULL,
+                             ctx->db_name,
+                             ctx->db_user,
+                             ctx->db_passwd);
+
+    if(PQstatus(ctx->conn) != CONNECTION_OK) {
+        flb_error("[out_pgsql] failed to connect to host=%s with error: %s",
+                  ctx->db_hostname, PQerrorMessage(ctx->conn));
+
+        pgsql_conf_destroy(ctx);
+
+        return -1;
+    }
+
+    flb_info("[out_pgsql] host=%s port=%s dbname=%s OK",
+              ctx->db_hostname, ctx->db_port, ctx->db_name);
+    flb_output_set_context(ins, ctx);
+
+    temp = PQescapeIdentifier(ctx->conn, ctx->db_table, flb_sds_len(ctx->db_table));
+
+    if(temp == NULL) {
+        flb_error("[out_pgsql] failed to parse table name: %s", PQerrorMessage(ctx->conn));
+        pgsql_conf_destroy(ctx);
+        return -1;
+    } else {
+        flb_sds_destroy(ctx->db_table);
+        ctx->db_table = flb_sds_create(temp);
```

Comment:
nit: this is the final value of db_table, the previous value comes from non-alloc'd sources (parameter or constant). So you could set that to a local variable in the function, and only create the allocation here. Would also improve the ease of understanding the code.

Reply:
Non nit: I do not see a check after this allocation.

Reply:
fixed!